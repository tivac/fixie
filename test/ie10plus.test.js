"use strict";

var simple = require("./util.js").simple,
    nested = require("./util.js").nested;

describe(":ie10plus", () => {
    it("should transform :ie10plus(...) into an ie10plus hack (single selector)", () =>
        simple(`
            :ie10plus(.fooga) {
                color: red;
            }
        `)
    );

    it("should transform :ie10plus(...) into an ie10plus hack (multiple selectors)", () =>
        simple(`
            :ie10plus(.fooga .wooga) {
                color: red;
            }
        `)
    );

    it("should work with postcss-nested", () =>
        nested(`
            .fooga {
                color: red;
                
                :ie10plus(&) {
                    color: blue;
                }
            }
        `)
    );
});
