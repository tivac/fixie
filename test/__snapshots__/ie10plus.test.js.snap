// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:ie10plus should transform :ie10plus(...) into an ie10plus hack (multiple selectors) 1`] = `
"_:-ms-lang(x),.fooga .wooga {
    color: red;
}"
`;

exports[`:ie10plus should transform :ie10plus(...) into an ie10plus hack (single selector) 1`] = `
"_:-ms-lang(x),.fooga {
    color: red;
}"
`;

exports[`:ie10plus should work with postcss-nested 1`] = `
".fooga {
    color: red;
}
    
    _:-ms-lang(x),.fooga {
        color: blue;
    }"
`;
